great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py

```diff
@@ -28,111 +30,248 @@ def plot(
+
+        column_domain_chart: alt.VConcatChart = (
+            self._create_chart_for_column_domain_expectations(
+                expectation_configurations=column_based_expectations,
+                attributed_metrics=attributed_metrics_by_column_domain,
+                prescriptive=prescriptive,
+            )
+        )
+        return column_domain_chart
 
-    def _plot_descriptive(
+    def _create_chart_for_column_domain_expectations(
         self,
-        df: pd.DataFrame,
-        metric_name: str,
-        metric_type: alt.StandardType,
-        domain_name: str,
-        domain_type: alt.StandardType,
-    ) -> alt.Chart:
-        descriptive_chart: alt.Chart = self.get_line_chart(
-            df=df,
+        expectation_configurations: List[ExpectationConfiguration],
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        prescriptive: bool,
+    ) -> alt.VConcatChart:
+        metric_name: Optional[str] = None
+        domain_name: str = "batch"
+        metric_type: str = AltairDataTypes.QUANTITATIVE.value
+        domain_type: str = AltairDataTypes.ORDINAL.value
+
+        dfs: List[Tuple[str, pd.DataFrame]] = []
+        for expectation_configuration in expectation_configurations:
+            metric_configuration: dict = expectation_configuration.meta[
+                "profiler_details"
+            ]["metric_configuration"]
+            domain_kwargs: dict = metric_configuration["domain_kwargs"]
+
+            domain: Domain = Domain(
+                domain_type=MetricDomainTypes.COLUMN,
+                domain_kwargs=domain_kwargs,
+            )
+            attributed_values_by_metric_name: Dict[
+                str, ParameterNode
+            ] = attributed_metrics[domain]
+
+            # Altair does not accept periods.
+            metric_name = list(attributed_values_by_metric_name.keys())[0].replace(
+                ".", "_"
+            )
+
+            df: pd.DataFrame = VolumeDataAssistantResult._create_df_for_charting(
+                metric_name, domain_name, attributed_values_by_metric_name
+            )
+
+            if prescriptive:
+                for kwarg_name in ("min_value", "max_value"):
+                    df[kwarg_name] = expectation_configuration.kwargs[kwarg_name]
+
+            column_name: str = expectation_configuration.kwargs["column"]
+            dfs.append((column_name, df))
+
+        assert metric_name is not None
+
+        plot_impl: Callable[
+            [
+                List[Tuple[str, pd.DataFrame]],
+                str,
+                alt.StandardType,
+                str,
+                alt.StandardType,
+            ],
+            alt.VConcatChart,
+        ]
+
+        if prescriptive:
+            plot_impl = self.get_expect_column_values_to_be_between_chart
+        else:
+            plot_impl = self.get_vertically_concatenated_chart
+
+        chart: alt.VConcatChart = plot_impl(
+            dfs_by_column=dfs,
```

Comment:
Maybe a rename to `column_dfs` is good enough?

Reply:
ah yup like that a lot more. thank you!